### PR TITLE
Add documentation for datatype updates

### LINKED ISSUE #243: No ability to change datatype on datasets
(Have not checked if this is an upstream API issue, just documenting the bug) cc @bebatut

### PR DESCRIPTION
**Pull request by simonbray**
To make it clear in the documentation that datatypes can be modified through bioblend.

Fixes #243.

### CODE REVIEW

**Comment by bgruening** [comment]
Cool, thanks!

**Comment by nsoranzo** [comment]
Thanks! Can you also add a test to `bioblend/_tests/TestGalaxyHistories.py` , limited to 20.09:

```python
@test_util.skip_unless_galaxy('release_20.09')
def test_update_dataset_datatype(self):
    ...
```

**Comment by nsoranzo** [comment]
Forgot to say, can you also please rebase on top of `master`? I've just added Travis testing on the (still unreleased) 20.09 branch.

File: bioblend/_tests/TestGalaxyHistories.py [hunk 1]
```diff
@@ -195,5 +195,18 @@ def test_copy_dataset(self):
...
+        dataset1_id = self._test_dataset(history_id)
+        original_hda = self.gi.datasets.show_dataset(dataset1_id)
+        self._wait_and_verify_dataset(dataset1_id, b'1\t2\t3\n')
+        assert original_hda['extension'] == 'auto'
+        assert original_hda['data_type'] == 'galaxy.datatypes.data.Data'
+        self.gi.histories.update_dataset(history_id, dataset1_id, datatype='tabular')
+        updated_hda = self.gi.datasets.show_dataset(dataset1_id)
+        assert updated_hda['extension'] == 'tabular'
+        assert updated_hda['data_type'] == 'galaxy.datatypes.tabular.Tabular'
+
```
Comment (simonbray) [inline bioblend/_tests/TestGalaxyHistories.py:210]: @nsoranzo, I kept it fairly simple here, in comparison to the Galaxy API test. If you would like to also see tests here for all the possible errors that can be raised, let me know.

Comment (nsoranzo) [reply bioblend/_tests/TestGalaxyHistories.py:210]: That's fine, thanks!

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Thanks @simonbray !

**Pull request merged**